theano/tensor/alt_gemm_template.c

```diff
@@ -0,0 +1,133 @@
+/** %(name)s **/
+void alt_numpy_scalar_matrix_product_in_place_%(float_type)s(%(float_type)s scalar, PyArrayObject* matrix) {
+    NpyIter* iterator = NpyIter_New(matrix, 
+        NPY_ITER_READWRITE | NPY_ITER_EXTERNAL_LOOP | NPY_ITER_REFS_OK, 
+        NPY_KEEPORDER, NPY_NO_CASTING, NULL);
+    if(iterator == NULL)
+        alt_fatal_error("Unable to iterate over a matrix "
+                        "for a scalar * matrix operation.");
+    NpyIter_IterNextFunc* get_next = NpyIter_GetIterNext(iterator, NULL);
+    char** data_ptr = NpyIter_GetDataPtrArray(iterator);
+    npy_intp* stride_ptr = NpyIter_GetInnerStrideArray(iterator);
+    npy_intp* innersize_ptr = NpyIter_GetInnerLoopSizePtr(iterator);
+    do {
+        char* data = *data_ptr;
+        npy_intp stride = *stride_ptr;
+        npy_intp count = *innersize_ptr;
+        while(count) {
+            *((%(float_type)s*)data) *= scalar;
+            data += stride;
+            --count;
+        }
+    } while(get_next(iterator));
+    NpyIter_Deallocate(iterator);
+}
+/*Matrix+Matrix function. Compute (coeffA * matrixA) + (coeffB * matrixB)
+ * Remark: This function actually sums a C-contiguous matrix (alpha*op(A)*op(B)) with a F-contiguous matrix (beta*C)
```

Comment:
Do you talk about the comment line ? If yes, I can remove it, but it is still true (that this function sums a C-contiguous with a F-contiguous matrix; I have made this comment to signal that there is a performance limitation here, as the iteration can't be as quickly as possible) !